src/BinSkim.Driver/RoslynAnalyzer/ILOperations.cs

```diff
@@ -0,0 +1,485 @@
+        public override ITypeSymbol ResultType { get; }
+        public override OperationKind Kind => OperationKind.ArrayCreationExpression;
+
+        IArrayInitializer IArrayCreationExpression.ElementValues => null;
+    }
+
+    internal sealed class AssignmentExpression : Expression, IAssignmentExpression
+    {
+        public AssignmentExpression(IReferenceExpression target, IExpression value)
+        {
+            Target = target;
+            Value = value;
+        }
+
+        public IReferenceExpression Target { get; }
+        public IExpression Value { get; }
+
+        public override OperationKind Kind => OperationKind.AssignmentExpression;
+        public override ITypeSymbol ResultType => Target.ResultType;
+    }
+
+    internal abstract class Statement : Operation, IStatement
+    {
+    }
+
+    internal sealed class ExpressionStatement : Statement, IExpressionStatement
+    {
+        public ExpressionStatement(IExpression expression)
+        {
+            Expression = expression;
+        }
+
+        public IExpression Expression { get; }
+        public override OperationKind Kind => OperationKind.ExpressionStatement;
+    }
+
+    internal class BlockStatement : Statement, IBlockStatement
+    {
+        public BlockStatement(ImmutableArray<ILocalSymbol> locals, ImmutableArray<IStatement> statements)
+        {
+            Locals = locals;
+            Statements = statements;
+        }
+
+        public ImmutableArray<ILocalSymbol> Locals { get; }
+        public ImmutableArray<IStatement> Statements { get; }
+        public override OperationKind Kind => OperationKind.BlockStatement;
+    }
+
+    // FEEDBACK: Why is there both ILabelStatement and ILabeledStatement? Not sure which to use.
+    internal sealed class LabelStatement : Statement, ILabelStatement
+    {
+        public LabelStatement(ILabelSymbol label)
+        {
+            Label = label;
+        }
+
+        public ILabelSymbol Label { get; }
+        public override OperationKind Kind => OperationKind.LabelStatement;
+    }
+
+    internal sealed class IfStatement : Statement, IIfStatement
+    {
+        public IfStatement(IExpression condition, IStatement ifTrue)
+        {
+            Condition = condition;
+            IfTrue = ifTrue;
+        }
+        public IExpression Condition { get; }
+        public IStatement IfTrue { get; }
+        public override OperationKind Kind => OperationKind.IfStatement;
+
+        IStatement IIfStatement.IfFalse => null; // we always goto on true and fallthrough on false. 
+    }
+
+    internal sealed class GoToStatement : Statement, IBranchStatement
+    {
+        public GoToStatement(ILabelSymbol target)
+        {
+            Target = target;
+        }
+
+        public ILabelSymbol Target { get; }
+
+        // FEEDBACK: first non 1:1 I see between interface name and kind, others have distinct subkinds. Should this be BranchKind.GoTo and OperationKind.BranchStatement?
```

Comment:
Interesting. I can imagine an analyzer caring about distinguishing between the various kinds of conditional branches--wanting to analyze goto statements but not exit statements, for example. So, yes, I think we should have BranchStatement and a BranchKind enum.